### PR TITLE
Fix gas-water initialization with transision zone

### PR DESCRIPTION
**Pull request by totto82**
I preliminary quick fix. Need to test a bit more before merging.

### CODE REVIEW

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
https://github.com/OPM/opm-simulators/issues/5279

**Comment by totto82** [comment]
jenkins build this please

**Comment by lisajulia** [comment]
I would like to include this into the release candidate 2, so I'll wait until this is merged to master until I create the new release candidate, ok?

**Comment by lisajulia** [comment]
> image image
> 
> I have tested this PR on a simple box model with CO2/BRINE for both the oil-gas and the water-gas model.
> 
> With
> 
> ```
> EQUIL
> 0.0 200 1000 0 5 0.0 1 0 0 /
> /
> ```
> 
> for the gas-oil case and
> 
> ```
> EQUIL
> 0.0 200 5 0.0 1* 0 0 0 0 /
> /
> ```
> 
> for the gas-water case.
> 
> And they agree and gives results are expected.

Is is needed / beneficial to add this test to opm-tests? :eyes: If not, then nevermind :).

File: opm/simulators/flow/equil/EquilibrationHelpers_impl.hpp [hunk 1]
```diff
@@ -448,6 +448,13 @@ operator()(double s) const
     fluidState.setSaturation(FluidSystem::gasPhaseIdx, 0.0);
     fluidState.setSaturation(phase_, s);
 
+    // This code should only be called for water or gas phase
+    assert(phase_ != FluidSystem::oilPhaseIdx);
+    // The capillaryPressure code only uses the wetting phase saturation
+    if (phase_ == FluidSystem::gasPhaseIdx) {
+        fluidState.setSaturation(FluidSystem::waterPhaseIdx, 1.0 - s);
```
Comment (atgeirr) [inline opm/simulators/flow/equil/EquilibrationHelpers_impl.hpp:455]: Should this not be 0? Or is this a small trick to handle both gas-water and gas-oil systems?

File: opm/simulators/flow/equil/InitStateEquil_impl.hpp [hunk 2]
```diff
@@ -578,9 +578,11 @@ deriveSaturations(const Position& x,
     this->setEvaluationPoint(x, reg, ptable);
     this->initializePhaseQuantities();
 
-    if (ptable.waterActive()) { this->deriveWaterSat(); }
     if (ptable.gasActive())   { this->deriveGasSat();   }
 
+    if (ptable.waterActive()) { this->deriveWaterSat(); }
```
Comment (atgeirr) [inline opm/simulators/flow/equil/InitStateEquil_impl.hpp:583]: Why is the order changed for the gasActive() and waterActive() related lines?

Nitpick: double blank line below.

**Review by atgeirr — COMMENTED** [review]
Just two questions, otherwise ok.

File: opm/simulators/flow/equil/EquilibrationHelpers_impl.hpp (hunk 1 above)
Comment (totto82) [reply opm/simulators/flow/equil/EquilibrationHelpers_impl.hpp:455]: Yes. It the later code will only use the wetting phase saturation. So water for gas-water and oil-water systems and oil for oil-gas systems.

File: opm/simulators/flow/equil/InitStateEquil_impl.hpp (hunk 2 above)
Comment (totto82) [reply opm/simulators/flow/equil/InitStateEquil_impl.hpp:583]: For 2p gas-water Sw = 1 - Sg. So we need to compute the gas saturation first.

**Comment by lisajulia** [comment]
@totto82, @atgeirr: Sorry for asking you once more about that: By when do you think this will be merged?
I'd like to create the tags for the release candidate 2 by tomorrow, and it would be great if we could have this in as well :)!
Thanks!

**Comment by totto82** [comment]
From my point of view this is ok. @atgeirr Please merge if you are happy with my comments.

**Pull request merged**